src/sql/src/plan/query.rs

```diff
@@ -147,39 +147,44 @@ pub fn plan_insert_query(
         bail!("cannot insert into system table '{}'", table.name());
     }
 
+    // Normalize column names
+    let columns: Vec<_> = columns.into_iter().map(normalize::column_name).collect();
+
     // Validate target column order.
-    let mut project_key: Vec<_> = (0..desc.arity()).collect();
-    let mut ordering = Vec::with_capacity(desc.arity());
+    let mut source_types = Vec::with_capacity(columns.len());
+    let mut ordering = Vec::with_capacity(columns.len());
 
     if columns.is_empty() {
+        // Columns in source query must be in order. Let's guess the full shape and truncate to the
```

Comment:
Same thought about the word `source` here! Maybe something like this would work: 
```
If columns are not explicitly included in the INSERT statement, guess the full shape ...
```